### PR TITLE
Add support for ASN.1 Printable string and UTF8 string values.

### PR DESCRIPTION
**Pull request by qmfrederik**
This add two new classes, `DerAsnPrintableString` and `DerAsnUtf8String`, which allow encoding/decoding of ASN.1 Printable String and UTF8 String values.

### CODE REVIEW

**Comment by huysentruitw** [comment]
Thanks for the pull-request invasion. Will check them one-by-one 👍

File: src/DerConverter/Asn/DerAsnPrintableString.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        public DerAsnPrintableString(byte[] bytes)
+            : base(DerAsnTypeTag.PrintableString)
+        {
+            if (bytes == null) throw new ArgumentNullException(nameof(bytes));
+            _bytes = bytes.ToList();
+        }
+
+        public override object Value => Encoding.ASCII.GetString(_bytes.ToArray());
+
+        public override string ToString()
```
Comment (huysentruitw) [inline src/DerConverter/Asn/DerAsnPrintableString.cs]: Please remove ToString override (if needed, we could add a ToString in the base-class, but this is out of scope of this PR).

File: src/DerConverter/Asn/DerAsnUtf8String.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        public DerAsnUtf8String(byte[] bytes)
+            : base(DerAsnTypeTag.Utf8String)
+        {
+            if (bytes == null) throw new ArgumentNullException(nameof(bytes));
+            _bytes = bytes.ToList();
+        }
+
+        public override object Value => Encoding.UTF8.GetString(_bytes.ToArray());
+
+        public override string ToString()
```
Comment (huysentruitw) [inline src/DerConverter/Asn/DerAsnUtf8String.cs]: Please remove.

File: src/DerConverter/Asn/DerAsnPrintableString.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace DerConverter.Asn
+{
+    public class DerAsnPrintableString : DerAsnType
```
Comment (huysentruitw) [inline src/DerConverter/Asn/DerAsnPrintableString.cs:8]: Please add a Parse and GetBytes unit-test for this type. See https://github.com/huysentruitw/pem-utils/blob/master/tests/DerConverter.Tests/Asn/DerAsnBitStringTests.cs .

File: src/DerConverter/Asn/DerAsnUtf8String.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace DerConverter.Asn
+{
+    public class DerAsnUtf8String : DerAsnType
```
Comment (huysentruitw) [inline src/DerConverter/Asn/DerAsnUtf8String.cs:8]: Please add a Parse and GetBytes unit-test for this type. See https://github.com/huysentruitw/pem-utils/blob/master/tests/DerConverter.Tests/Asn/DerAsnBitStringTests.cs .

**Review by huysentruitw — CHANGES_REQUESTED** [review]

**Comment by qmfrederik** [comment]
Removed the `ToString` overloads and added unit tests.

File: tests/DerConverter.Tests/Asn/DerAsnPrintableStringTests.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+using NUnit.Framework;
+using System.Linq;
+
+namespace DerConverter.Tests.Asn
+{
+    [TestFixture]
+    public class DerAsnPrintableStringTests
+    {
+        [Test]
+        public void DerAsnPrintable_Parse_ShouldDecodeCorrectly()
```
Comment (huysentruitw) [inline tests/DerConverter.Tests/Asn/DerAsnPrintableStringTests.cs]: `DerAsnPrintableString_Parse_ShouldDecodeCorrectly`

File: tests/DerConverter.Tests/Asn/DerAsnPrintableStringTests.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            };
+
+            var type = DerAsnPrintableString.Parse(data);
+            Assert.That(type is DerAsnPrintableString, Is.True);
+
+            Assert.That(type.Value, Is.EqualTo(data.Skip(2).ToArray()));
+        }
+
+        [Test]
+        public void DerAsnPrintable_GetBytes_ShouldEncodeCorrectly()
```
Comment (huysentruitw) [inline tests/DerConverter.Tests/Asn/DerAsnPrintableStringTests.cs]: `DerAsnPrintableString_GetBytes_ShouldEncodeCorrectly`

**Review by huysentruitw — CHANGES_REQUESTED** [review]

File: tests/DerConverter.Tests/Asn/DerAsnPrintableStringTests.cs (hunk 6 above)
Comment (qmfrederik) [reply tests/DerConverter.Tests/Asn/DerAsnPrintableStringTests.cs]: Oh, good catch. Fixed it.

**Review by huysentruitw — APPROVED** [review]
